src/internal/util/isReadableStreamLike.ts

```diff
@@ -0,0 +1,34 @@
+import { isFunction } from './isFunction';
+
+// We shouldn't rely on this type definition being available globally yet since it's
+// not necessarily available in every TS environment.
+interface ReadableStreamDefaultReaderLike<T> {
+  read(): IteratorResult<T>;
+  releaseLock(): void;
+}
+
+// This is the only part that matters to us.
+export interface ReadableStreamLike<T> {
+  getReader(): ReadableStreamDefaultReaderLike<T>;
+}
+
+export async function* readableStreamLikeToAsyncGenerator<T>(readableStream: ReadableStreamLike<T>): AsyncGenerator<T> {
```

Comment:
Just thinking... that while this is _definitely_ the coolest way. The lame way might be more efficient, something like:

```ts
export function readableStreamLikeToAsyncGenerator(readableStream: ReadableStreamLike<T>): AsyncGenerator<T> {
  const reader = readableStream.getReader();
  
  return {
     [Symbol.asyncIterator]() { return this; },
     async next() { 
       return reader.read();
     },
     async throw(err) {
       reader.releaseLock();
       throw err;
     },
     async return() {
       reader.releaseLock();
       return { done: true };
     }
  }
}
```

But maybe I'm totally wrong. Just something I was thinking about. I guess if there was an error during the execution of `next` above, the lock wouldn't get released unless you had a try/catch in there, so maybe this isn't the best idea.